manifests/init.pp

```diff
@@ -945,6 +1078,34 @@
     $my_pam_session_lines = $pam_session_lines
   }
 
+  if $pam_password_auth_lines == undef {
+    $my_pam_password_auth_lines = $default_pam_password_auth_lines
+  } else {
+    $my_pam_password_auth_lines = $pam_password_auth_lines
+    validate_array($my_pam_password_auth_lines)
```

Comment:
@nbentoumi - Wouldn't we just want to validate the $my_pam_password*lines array after each if statement.  That way you validate the default and user input?